src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public void registerUser(String email, String password) throws RegistrationException {
+        Optional<User> optionalUserByEmail = userService.findByEmail(email);
+        if (optionalUserByEmail.isPresent()) {
+            throw new RegistrationException("Email " + email + "already registered");
+        }
+        User newUser = new User();
+        newUser.setEmail(email);
+        newUser.setPassword(password);
```

Comment:
Ensure that the password is hashed using `HashUtil.hashPassword(password, salt)` before setting it. Also, generate a salt for the user.